#! /usr/bin/env python

PACKAGE = "robotx_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("linear_kp", double_t, 0, "linear kp ", 0.5, -10, 10.0)
gen.add("linear_ki", double_t, 0, "linear ki ", 0, -10, 10.0)
gen.add("linear_kd", double_t, 0, "linear kd ", 2.5, -10, 10.0)
gen.add("angular_kp", double_t, 0, "angular kp ", 0.7, -10, 10.0)
gen.add("angular_ki", double_t, 0, "angular ki ", 0.001, -10, 10.0)
gen.add("angular_kd", double_t, 0, "angular kd ", 0.07, -10, 10.0)

gen.add("linear_threshold", double_t, 0, "linear threshold", 5, 0, 10.0)
gen.add("linear_velocity_threshold", double_t, 0, "linear velocity threshold", 2, 0, 3)
gen.add("angular_velocity_threshold", double_t, 0, "angular velocity threshold", 0.5, 0, 3.14)

exit(gen.generate(PACKAGE, "robotx_control", "CmdVelPID"))
